extensions/measurement-tracking/src/viewports/TrackedCornerstoneViewport.js

```diff
@@ -68,6 +70,64 @@ function TrackedCornerstoneViewport({
     };
   }, []);
 
+  useEffect(() => {
+    const { unsubscribe } = MeasurementService.subscribe(
+      MeasurementService.EVENTS.JUMP_TO_MEASUREMENT,
+      ({ viewportIndex: jumpToMeasurementViewportIndex, measurement }) => {
+        // check if the correct viewport index.
+        // TODO -> Jump to measurement implementation.
+
+        if (viewportIndex !== jumpToMeasurementViewportIndex) {
+          // Event for a different viewport.
+          return;
+        }
+
+        if (
+          measurement.displaySetInstanceUID !== displaySet.displaySetInstanceUID
+        ) {
+          // Not for this displaySet.
+          return;
+        }
+
+        _jumpToMeasurement(
+          measurement,
+          element,
+          viewportIndex,
+          MeasurementService,
+          DisplaySetService
+        );
+      }
+    );
+
+    // Check if there is a queued jumpToMeasurement event
+    const measurementIdToJumpTo = MeasurementService.getJumpToMeasurement(
+      viewportIndex
+    );
```

Comment:
Ah. Here is the check, and it's in the useEffect w/ `displaySet` as a change check. That makes sense. Can we pull this out to make it super easy to read?

```js
useEffect(() => {
  const unsubcribeFromJumpToMeasurementEvents = subscribeToJumpToMeasurementEvents();
  
  checkForCachedJumpToMeasurementEvents();

 return () => {
   unsubcribeFromJumpToMeasurementEvents();
 }
}, [element, displaySet]);
```

Reply:
When component's start to get large, it's a PITA to understand what's going on (in my opinion). My hope is, that at  glance, I can see the intent, and then dig into behavior as needed.

The actual logic can live further down in this file, or in separate files that are imported (which also makes them easier to test, if we ever go that route.